### PR TITLE
Added shelter and gate occupation icon and animation

### PR DESCRIPTION
**Pull request by Zed9876**
[RainmeadowShelterGateIcon (2).zip](https://github.com/user-attachments/files/17708593/RainmeadowShelterGateIcon.2.zip)

Shelters and gates will now make slugcats permanently display icons that correlate to the type of room they are in

### CODE REVIEW

**Comment by Zed9876** [comment]
Just realised that the video was cut short at the beggining for some reason. find a better one here.
[ShelterAndGateIcon.zip](https://github.com/user-attachments/files/17708649/ShelterAndGateIcon.zip)

**Review by youbitchoc — CHANGES_REQUESTED** [review]
logic lgtm but pls make it more elegant

**Comment by youbitchoc** [comment]
just watched the vid and i think the offset flashing looks weird

**Comment by youbitchoc** [comment]
yeah if everything flashes together we could simplify the code

File: Story/OnlineUIComponents/OnlinePlayerDisplay.cs [hunk 1]
```diff
@@ -161,44 +192,92 @@ public override void Draw(float timeStacker)
...
-            this.arrowSprite.color = lighter_color;
-            this.slugIcon.color = lighter_color;
+                this.arrowSprite.color = lighter_color;
+                this.slugIcon.color = lighter_color;
+
+                if (resetUsernameCounter < 10) // snappier fadeaway
+                {
+                    this.username.color = lighter_color;
+
+                }
```
[the remark this answers is not in the document]
Comment (Zed9876) [reply Story/OnlineUIComponents/OnlinePlayerDisplay.cs:273]: Not sure, it was in the code before I touched it, i moved it to where it currently is, but I'll try removing it and see what happens.

**Comment by Zed9876** [comment]
Awesome, I'll get on the changes asap! (＾＾)ｂ

File: Story/OnlineUIComponents/OnlinePlayerDisplay.cs [hunk 2]
```diff
@@ -161,44 +164,59 @@ public override void Draw(float timeStacker)
...
+                {
+                    notificationTimer = this.notifyPlayer;
+                    if (this.flippedColor)
+                    {
+                        if (this.message.text == "") this.username.color = Color.white;
+                        this.slugIcon.color = Color.white;
+                        this.arrowSprite.color = Color.white;
+                        this.flippedColor = false;
+                    }
+                    else if (!this.flippedColor)
```
Comment (youbitchoc) [inline Story/OnlineUIComponents/OnlinePlayerDisplay.cs:198]: this if part is redundant, also consider flipping colour outside with foo = !foo

Comment (youbitchoc) [reply Story/OnlineUIComponents/OnlinePlayerDisplay.cs:198]: see snippet at https://github.com/henpemaz/Rain-Meadow/pull/444#discussion_r1837347544

**Comment by Zed9876** [comment]
I have found a few bugs during some more testing I've done today, will try and patch them asap.
Bugs found:
- Host Icon not showing (fixed)
- chat text showing above players' heads permanently


Will update as I progress

**Review by 6fears7 — APPROVED** [review]

**Pull request merged**